File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        
        Map<String, Integer[]> key = new HashMap<>();
        key.put("up", new Integer[] {0, 1});
        key.put("down", new Integer[] {0, -1});
        key.put("left", new Integer[] {-1, 0});
        key.put("right", new Integer[] {1, 0});
        
        int w = (board[0]-1)/2; 
        int h = (board[1]-1)/2; 
        
        for (String input : keyinput) {
            Integer[] k = key.get(input); 
            
            if (-w <= answer[0]+k[0] && answer[0]+k[0] <= w) {
                answer[0] += k[0];    
            }
            if (-h <= answer[1]+k[1] && answer[1]+k[1] <= h) {
                answer[1] += k[1];    
            }
        }
        
        return answer;
    }
}


    

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    w = (board[0]-1)//2
    h = (board[1]-1)//2
        
    for key in keyinput:
        if key == 'up' and answer[1] < h:
            answer[1] += 1                
        if key == 'down' and answer[1] > -h:
            answer[1] -= 1                
        if key == 'left' and answer[0] > -w:
            answer[0] -= 1
        if key == 'right' and answer[0] < w:
            answer[0] += 1
            
    return answer